lib/spack/spack/package_base.py

```diff
@@ -926,6 +922,32 @@ def global_license_file(self):
             self.global_license_dir, self.name, os.path.basename(self.license_files[0])
         )
 
+    # NOTE: return type should be Optional[Literal['all', 'specific', 'none']] in
```

Comment:
So... I was really, really wanting typing_extensions so I could fix typing information for all of our decorators the other day, and it has this.  Not in this issue, but lets vendor that shall we?

https://typing-extensions.readthedocs.io/en/latest/